projects/geo/src/lib/layer/layer-legend/layer-legend.component.ts

```diff
@@ -27,5 +28,25 @@ export class LayerLegendComponent {
   }
   private _legend;
 
-  constructor() {}
+  constructor(private capabilitiesService: CapabilitiesService) {}
+
+  validateItemTitle(item): string {
+    const layerOptions = this.layer.dataSource.options as any;
+    if (layerOptions.type === 'wms' && layerOptions.optionsFromCapabilities) {
+      let localLayerOptions = JSON.parse(JSON.stringify(layerOptions)); // to avoid to alter the original options.
+      layerOptions.params.layers.split(',').forEach(layer => {
+        if (layer === item.title) {
+          localLayerOptions.params.layers = layer;
+          this.capabilitiesService.getWMSOptions(localLayerOptions).subscribe(r => localLayerOptions = r);
+        }
+      });
+      if (localLayerOptions._layerOptionsFromCapabilities && localLayerOptions._layerOptionsFromCapabilities) {
```

Comment:
Why the && in this condition?

Reply:
Hum. A wrong copy&paste? 

```suggestion
      if (localLayerOptions && localLayerOptions._layerOptionsFromCapabilities) {
```